async/delay.ts

```diff
@@ -3,11 +3,13 @@
 
 export interface DelayOptions {
   signal?: AbortSignal;
+  /** Defines if the timer should be non-blocking. This is `false` by default. */
```

Comment:
`the timer should be non-blocking` sounds strange to me as timers are always async operations. I would document this like `Indicates whether the process can exit while the timer still existing`

Reply:
> Another naming candidate for this option would be `persistent` (with opposite effect of boolean value) as used in `fs.watch` of node.js https://nodejs.org/api/fs.html#fswatchfilename-options-listener
> 
> Which do you think is better?

Yeah, I like persistent.